test/people-route-test.js

```diff
@@ -0,0 +1,204 @@
+          expect(res.status).to.equal(400);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('GET: /api/people/:id', function() {
+    describe('with a valid body', function() {
+      before( done => {
+        new People(examplePeople).save()
+        .then( people => {
+          this.tempPeople = people;
+          return People.findByIdAndAddPresident(people._id, examplePresident)
+        })
+        .then( president => {
+          this.tempPresident = president;
+          done();
+        })
+        .catch(done);
+      });
+
+      after( done => {
+        if(this.tempPeople) {
+          People.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        };
+        done();
+      });
+
+      it('should return an actual people', done => {
+        request.get(`${url}/api/people/${this.tempPeople._id}`)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.name).to.equal('test people name');
+          expect(res.body.presidents.length).to.equal(1);
+          done();
+        });
+      });
+    });
+
+    describe('with an invalid id', function() {
+      it('should return a 404 error', function() {
+        request.get(`${url}/api/people/121212`)
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('PUT: api/people/:id', function() {
+    describe('with a valid id and body', function() {
+      before( done => {
+        new People(examplePeople).save()
+        .then( people => {
+          this.tempPeople = people;
+          done();
+        })
+        .catch(done);
+      });
+
+      after( done => {
+        if(this.tempPeople) {
+          People.remove({})
+          .then(() => done())
+          .catch(done);
+          return;
+        };
+        done();
+      });
+
+      it('should update a people with a valid id and body', done => {
+        let updatePeople = {name: 'some new people'}
+        request.put(`${url}/api/people/${this.tempPeople._id}`)
+        .send(updatePeople)
+        .end((err, res) => {
+          if(err) return done(err);
+          let timestamp = new Date(res.body.timestamp);
+          expect(res.status).to.equal(200);
+          expect(timestamp.toString()).to.equal(examplePeople.timestamp.toString());
+          expect(res.body.name).to.equal(updatePeople.name);
+          done();
+        });
+      });
+    });
+
+    describe('with an invalid id', function() {
+      it('should return a 404 error with an invalid id', function() {
+        request.put(`${url}/api/people/28288282`)
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+      });
+    });
+
+    describe('with no body supplied', function() {
+      it('should return a 400 error', function() {
+        request.put(`${url}/api/people`)
```

Comment:
An alternative way to possibly write this test would be to pass a valid ID and then an invalid param to update the object with. For example, instead of passing `{name: 'some name'}` you could pass `{ nam: 'misspelled name'}`